File: frontend/lib/models/report.dart
class Report {
  final int id;
  final int conversationId;
  final List<Highlight> highlights;
  final List<SentimentAnalysis> sentimentAnalysis;
  final List<Suggestion> suggestions;
  final int pronunciationScore;
  final int grammarScore;
  final int fluencyScore;
  final int overallScore;
  final DateTime generatedAt;

  Report({
    required this.id,
    required this.conversationId,
    required this.highlights,
    required this.sentimentAnalysis,
    required this.suggestions,
    required this.pronunciationScore,
    required this.grammarScore,
    required this.fluencyScore,
    required this.overallScore,
    required this.generatedAt,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    var highlightsList = json['highlights'] as List;
    List<Highlight> highlights = highlightsList.map((h) => Highlight.fromJson(h)).toList();

    var sentimentList = json['sentiment_analysis'] as List;
    List<SentimentAnalysis> sentimentAnalysis = sentimentList.map((s) => SentimentAnalysis.fromJson(s)).toList();

    var suggestionsList = json['suggestions'] as List;
    List<Suggestion> suggestions = suggestionsList.map((s) => Suggestion.fromJson(s)).toList();

    return Report(
      id: json['id'],
      conversationId: json['conversation_id'],
      highlights: highlights,
      sentimentAnalysis: sentimentAnalysis,
      suggestions: suggestions,
      pronunciationScore: json['pronunciation_score'],
      grammarScore: json['grammar_score'],
      fluencyScore: json['fluency_score'],
      overallScore: json['overall_score'],
      generatedAt: DateTime.parse(json['generated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'conversation_id': conversationId,
      'highlights': highlights.map((h) => h.toJson()).toList(),
      'sentiment_analysis': sentimentAnalysis.map((s) => s.toJson()).toList(),
      'suggestions': suggestions.map((s) => s.toJson()).toList(),
      'pronunciation_score': pronunciationScore,
      'grammar_score': grammarScore,
      'fluency_score': fluencyScore,
      'overall_score': overallScore,
      'generated_at': generatedAt.toIso8601String(),
    };
  }
}

class Highlight {
  final String text;
  final DateTime timestamp;
  final String translation;

  Highlight({
    required this.text,
    required this.timestamp,
    required this.translation,
  });

  factory Highlight.fromJson(Map<String, dynamic> json) {
    return Highlight(
      text: json['text'],
      timestamp: DateTime.parse(json['timestamp']),
      translation: json['translation'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'text': text,
      'timestamp': timestamp.toIso8601String(),
      'translation': translation,
    };
  }
}

class SentimentAnalysis {
  final DateTime timestamp;
  final double sentimentScore;
  final String text;

  SentimentAnalysis({
    required this.timestamp,
    required this.sentimentScore,
    required this.text,
  });

  factory SentimentAnalysis.fromJson(Map<String, dynamic> json) {
    return SentimentAnalysis(
      timestamp: DateTime.parse(json['timestamp']),
      sentimentScore: json['sentiment_score'].toDouble(),
      text: json['text'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'timestamp': timestamp.toIso8601String(),
      'sentiment_score': sentimentScore,
      'text': text,
    };
  }
}

class Suggestion {
  final String category;
  final String text;
  final int importance;

  Suggestion({
    required this.category,
    required this.text,
    required this.importance,
  });

  factory Suggestion.fromJson(Map<String, dynamic> json) {
    return Suggestion(
      category: json['category'],
      text: json['text'],
      importance: json['importance'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'category': category,
      'text': text,
      'importance': importance,
    };
  }
}

File: backend/app/models/report.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Float
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base

class Report(Base):
    __tablename__ = "reports"
    
    id = Column(Integer, primary_key=True, index=True)
    conversation_id = Column(Integer, ForeignKey("conversations.id"))
    highlights = Column(Text)  # JSON格式存储高亮金句
    sentiment_analysis = Column(Text)  # JSON格式存储情感分析数据
    suggestions = Column(Text)  # JSON格式存储改进建议
    pronunciation_score = Column(Float, nullable=True)  # 发音评分
    grammar_score = Column(Float, nullable=True)  # 语法评分
    fluency_score = Column(Float, nullable=True)  # 流利度评分
    vocabulary_score = Column(Float, nullable=True)  # 词汇评分
    overall_score = Column(Float, nullable=True)  # 综合评分
    generated_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # 关联关系 - 使用字符串引用解决循环依赖
    conversation = relationship("Conversation", back_populates="report")

class HighlightPhrase(Base):
    __tablename__ = "highlight_phrases"
    
    id = Column(Integer, primary_key=True, index=True)
    report_id = Column(Integer, ForeignKey("reports.id"))
    text = Column(Text)
    timestamp = Column(DateTime(timezone=True))
    translation = Column(Text, nullable=True)

class SentimentPoint(Base):
    __tablename__ = "sentiment_points"
    
    id = Column(Integer, primary_key=True, index=True)
    report_id = Column(Integer, ForeignKey("reports.id"))
    timestamp = Column(DateTime(timezone=True))
    sentiment_score = Column(Float)  # -1 到 1，-1表示负面，1表示正面
    text = Column(Text)

class ImprovementSuggestion(Base):
    __tablename__ = "improvement_suggestions"
    
    id = Column(Integer, primary_key=True, index=True)
    report_id = Column(Integer, ForeignKey("reports.id"))
    category = Column(String)  # 改进建议类别，如"词汇"、"语法"、"发音"
    text = Column(Text)  # 改进建议内容
    importance = Column(Integer)  # 重要性，1-5级